cluster/manifests/z-karpenter/karpenter.k8s.aws_awsnodetemplates.yaml

```diff
@@ -197,54 +197,17 @@ spec:
               networkInterfaces:
                 items:
                   properties:
-                    associateCarrierIPAddress:
-                      description: "Associates a Carrier IP address with eth0 for
-                        a new network interface. \n Use this option when you launch
-                        an instance in a Wavelength Zone and want to associate a Carrier
-                        IP address with the network interface. For more information
-                        about Carrier IP addresses, see Carrier IP addresses (https://docs.aws.amazon.com/wavelength/latest/developerguide/how-wavelengths-work.html#provider-owned-ip)
-                        in the Wavelength Developer Guide."
-                      type: boolean
                     associatePublicIPAddress:
                       description: Associates a public IPv4 address with eth0 for
                         a new network interface.
                       type: boolean
-                    deleteOnTermination:
```

Comment:
why these fields are removed?

Reply:
I rebased the patch on your simplified PR where these fields were dropped: https://github.com/aws/karpenter-provider-aws/pull/4353

The other fields we don't use so it doesn't matter IMO, but `deleteOnTermination` I didn't think about if it breaks behavior :thinking:

Reply:
I will test it before we merge :) If it does delete them, then we're good IMO, if it leaks, then it's not ok.

Reply:
Succesfully tested, you're right the default is `deleteOnTermination=true`

![image](https://github.com/zalando-incubator/kubernetes-on-aws/assets/128566/a5d37d53-7a72-41fd-af63-9734003d8069)

So it's not leaking anything :tada: